kpi/permissions.py

```diff
@@ -121,6 +120,38 @@ def has_object_permission(self, request, view, obj):
         return True
 
 
+class AssetPermission(permissions.DjangoObjectPermissions):
+
+    # Setting this to False allows real permission checking on AnonymousUser.
+    # With the default of True, anonymous requests are categorically rejected.
+    authenticated_users_only = False
+
+    perms_map = permissions.DjangoObjectPermissions.perms_map.copy()
+    perms_map['GET'] = ['%(app_label)s.view_%(model_name)s']
+    perms_map['OPTIONS'] = perms_map['GET']
+    perms_map['HEAD'] = perms_map['GET']
+
+    def has_object_permission(self, request, view, obj):
+
+        user = get_database_user(request.user)
+        # When calling the endpoint with the API renderer, it calls this method
+        # for each method declared in `perms_maps`. To detect the real method,
+        # we need to access `request._request.method`
+        method = request._request.method
+        if (
+            method == 'PATCH'
+            and user_has_project_view_asset_perm(
+                obj, user, PERM_CHANGE_METADATA_ASSET
+            )
+        ) or (
+            method == 'GET'
+            and user_has_project_view_asset_perm(obj, user, PERM_VIEW_ASSET)
+        ):
+            return True
```

Comment:
We discussed how this gets [properly limited](https://github.com/kobotoolbox/kpi/blob/dc615657b24f10287ea7ab186a1c3710d07185d2/kpi/serializers/v2/asset.py#L180-L192) in the serializer, but we should add a test for that

Reply:
Done in https://github.com/kobotoolbox/kpi/pull/4225/commits/ff12597fa445e07d2acf0415c48292c927d96802